core/server/api/canary/utils/serializers/output/utils/mapper.js

```diff
@@ -57,6 +58,17 @@ const mapPost = (model, frame) => {
         });
     }
 
+    // Transforms post/page metadata to flat structure
+    let postsMeta = Object.assign({}, _.mapValues(postsMetaSchema, () => null), jsonModel.posts_meta);
```

Comment:
For the sake of keeping the syntax consistent maybe worth doing similar transformation to the one in the input serializer:
```
let metaAttrs = _.keys(_.omit(postsMetaSchema, ['id', 'post_id']));
```

This would avoid doing `delete` statements afterward.

Reply:
In addition, `metaAttrs` could be filtered with `!frame.options.columns || (frame.options.columns && frame.options.columns.includes(k)))` so that later on it's just a flat assignment to jsonMode :wink:

Reply:
@gargol Thanks, good suggestion, have cleaned up the code in mappers and it looks much cleaner now. lmk if this looks good :)